tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-steps-selector.html

```diff
@@ -136,56 +134,29 @@
     },
     _sliderValueChanged(event) {
       const run = event.target.dataset.run;
-      const newRunToStepIndex = _.clone(this._runToStepIndex);
-      newRunToStepIndex[run] = event.target.immediateValue;
-      this.set('_runToStepIndex', newRunToStepIndex);
+      this.set(`_runToStepIndex.${run}`, event.target.immediateValue);
     },
     _computeMaxStepIndexForRun(runToAvailableTimeEntries, run) {
       const entries = runToAvailableTimeEntries[run];
       return entries && entries.length ? entries.length - 1 : 0;
     },
-    _updateStepsForNewRuns(runs, runToAvailableTimeEntries, runToStepIndex) {
+    _updateStepsForNewRuns(runToAvailableTimeEntries) {
       // The mapping from run to available time entries just changed.
-      const newRunToStepIndex = {};
-      let runAdded = false;
       _.forOwn(runToAvailableTimeEntries, (entries, run) => {
-        if (!_.isNumber(runToStepIndex[run])) {
+        if (!_.isNumber(this._runToStepIndex[run])) {
           // This run had previously lacked a slider. Initially set the slider
           // for the run to point to the last step.
-          newRunToStepIndex[run] = entries.length - 1;
-          runAdded = true;
-        } else {
-          // The step for this run did not change.
-          newRunToStepIndex[run] = runToStepIndex[run];
+          this.set(`_runToStepIndex.${run}`, entries.length - 1);
```

Comment:
Does this work with runs names like `foo.baz.bar`? This can be common
for TensorBoard runs that encode hyperparameter values in the run name,
like `train/lr=0.01;dropout=0.2`.

(If so, how does it work?)

Reply:
Thanks a lot for catching that. Yeah that wouldn't work and I now want to do a quick audit on where we use `this.set`.
Addressed it by using `Object.assign`.